Cython/Compiler/TypeStubGenerator.py

```diff
@@ -0,0 +1,433 @@
+    def translate_base_type_to_py(
+        self,
+        base
+        ):
+
+        # Try checking our table first...
+        if self.translation_table.get(base.name):
+            return self.translation_table[base.name]
+
+        elif base.name == "object":
+            return "object"
+
+        elif base.name in ("unicode","basestring"):
+            return "str"
+
+        elif not base.is_basic_c_type:
+            # Likely that it's already a python object that's being handled...
+            # except for basestring and unicode...
+            return base.name 
+
+        elif base.name == "bint":
+            return "bool"
+
+        ctype = PyrexTypes.simple_c_type(base.signed, base.longness, base.name) # type: ignore
+        return self.translate_pyrex_type(ctype)
+
+    def emptyline(self):
+        self.result.putline("")
+    
+    def visit_ModuleNode(self, node):
+        # We need to extract the name to write our pyi file down...
+        if node.directives['write_stub_file']:
+            result = self.write(node, True)
+            new_path = node.full_module_name.replace(".", "/")
+            print("writing file %s.pyi ..." % node.full_module_name)
+            with open_new_file(os.path.join(new_path + '.pyi')) as w:
+                w.write("\n".join(result.lines))
+                w.write("\n")
+        return node
+
+    def visit_StatListNode(self, node):
+        self.visitchildren(node)
+        return node
+
+    def visit_CImportStatNode(self,node):
+        return node
+    
+    def visit_FromCImportStatNode(self,node):
+        return node
+    
+    def visit_CDefExternNode(self,node):
+        self.visitchildren(node)
+        return node 
+
+    def visit_CEnumDefNode(self, node):
+        # TODO Figure out how to define an enum-class via typehints...
+
+        # NOTE It seems that only public will make the enum accessible to python so 
+        # I'll just have it check if the enums will be public for now... - Vizonex
+        if node.visibility == "public":
+            # Enum's name is not in or visible in the final product because 
+            # it's not an enum class so do not indent here...
+            # Also Leave visit_CEnumDefItemNode up to the previous 
+            # class's function...
+            self.putline("# -- enum %s --" % node.name)
+            self.visitchildren(node)
+        return node 
+
+    # Used in our translation table to register return types variables from...
+    def visit_CTypeDefNode(self,node):
+        if isinstance(node.declarator, CNameDeclaratorNode):
+            # Register a new type to use in our translation table...
+            self.translation_table[node.declarator.name] = self.translate_base_type_to_py(node.base_type)
+    
+    def visit_CStructOrUnionDefNode(self, node):
+        # XXX : Currently, I don't know what to do here yet but ignoring 
+        # is triggering some problems currently...
+        return node
```

Comment:
We may consider mapping these to a namedtuple or dataclass.

Reply:
I probably wouldn't map them to anything for the first iteration at least - I think the main use case is helping editors know about things imported from Cython modules, and since cstructs and unions aren't importable into Python they're probably best left for now.